pkg/dataobj/internal/sections/logs/logs.go

```diff
@@ -21,10 +22,35 @@ import (
 
 // A Record is an individual log record within the logs section.
 type Record struct {
-	StreamID  int64
-	Timestamp time.Time
-	Metadata  labels.Labels
-	Line      []byte
+	StreamID    int64
+	Timestamp   time.Time
+	Metadata    labels.Labels
+	Line        []byte
+	MdValueCaps []int
+}
+
+func (r *Record) DeepCopy() Record {
```

Comment:
I'm not sure if I should have this or not. It's supposed to be a helper to a caller can copy elements from the buffer they passed in and can re-use the buffer. Maybe a better solution is to create new buffers from the client when something is returned and never pass the same buffer into a Read call?